pkg/verification/verification_service.go

```diff
@@ -57,28 +60,72 @@ func NewVerificationService(cfg ServiceConfiguration, opts ...VerificationServic
 	return s
 }
 
-// SendVerification sends a verification message to the specified user.  If successful, it will modify the
-// specified UserSignup parameter, setting the verification code annotation label to the value of the
-// generated verification code
-func (s *ServiceImpl) SendVerification(ctx *gin.Context, signup *v1alpha1.UserSignup) error {
-	verificationCode, err := generateVerificationCode()
+// InitVerification sends a verification message to the specified user.  If successful,
+// the user will receive a verification SMS.
+func (s *ServiceImpl) InitVerification(ctx *gin.Context, signup *v1alpha1.UserSignup, countryCode, phoneNumber string) (*v1alpha1.UserSignup, error) {
+	// get phone number hash
+	md5hash := md5.New()
+	// Ignore the error, as this implementation cannot return one
+	_, _ = md5hash.Write([]byte(countryCode + phoneNumber))
+	phoneHash := hex.EncodeToString(md5hash.Sum(nil))
+
+	signup.Labels[v1alpha1.UserSignupUserPhoneHashLabelKey] = phoneHash
+
+	now := time.Now()
+
+	// If 24 hours has passed since the verification timestamp, then reset the timestamp and verification attempts
+	ts, err := time.Parse(TimestampLayout, signup.Annotations[v1alpha1.UserSignupVerificationInitTimestampAnnotationKey])
+	if err != nil || (err == nil && now.After(ts.Add(24*time.Hour))) {
+		// Set a new timestamp
+		signup.Annotations[v1alpha1.UserSignupVerificationInitTimestampAnnotationKey] = now.Format(TimestampLayout)
+		signup.Annotations[v1alpha1.UserSignupVerificationCounterAnnotationKey] = "0"
+	}
+
+	// get the annotation counter
+	annotationCounter := signup.Annotations[v1alpha1.UserSignupVerificationCounterAnnotationKey]
+	var counter int
+	if annotationCounter == "" {
+		counter = 0
```

Comment:
Does `counter` need to be set here?  It seems that in just a few lines down (line 92) it will be overridden anyway.